### DISCUSSION TITLE
cannot change name of sql file

### DISCUSSION THREAD

**Discussion by selster1** [dbeaver/dbeaver#35174] [Q&A] [answered]
### Description

I am attaching 2 files, one shows a tab that will allow the change of an sql file, the other tab will not allow for renaming. Thanks, Steve

### DBeaver Version

24.1.2.202407071610

### Operating System

win 10 64 bit

### Steps to reproduce

when I edit sql scripts, sometimes the tab will allow me to change the name of the sql editor file, sometimes not, one example of each ia attached:

![no rename Clipboard01](https://github.com/dbeaver/dbeaver/assets/175140590/ad20730e-48be-4303-af1c-96bb7bca8a6d)

![Clipboard01](https://github.com/dbeaver/dbeaver/assets/175140590/a1274af9-7e4d-4c8f-81d1-dc62f9fe5e44)

**Comment by E1izabeth** [comment]
It might happen, if the script was not added to the DBeaver project.
Can you locate this script if you switch to the Projects view and press Link with editor, when the script is opened?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/341e77ae-f2ec-422c-93fa-dc79809869b2)

This button should jump to the active script.

**Comment by selster1** [comment]
ctrl-shift + ,  does nothing.  Also I do no know where this options is? "Projects view and press Link with editor,"  I do see icons for Project Explorer and for Projects, but I do not see options for "Projects view" and there is not choice to press Link with editor.  When is this happening.  Sometimes I brush against a key (I have not figured out which one yet) and the sql tab will instantly close, without asking me if I really want to close it.  Then I have to find the sql file, re- open in, but then I cannot rename it.  Thanks

**Comment by serjiokov** [comment, marked as answer]
@selster1 Seems you have an external script resource that is not related to any project, 
Please link or drop it to the script location folder
![image](https://github.com/user-attachments/assets/50b8ebfd-9eaf-4337-962b-cced615325c3)
